server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -59,15 +80,34 @@ public Query existsQuery(SearchExecutionContext context) {
         public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
             return new StoredValueFetcher(context.lookup(), NAME);
         }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            if (hasDocValues() == false) {
+                throw new IllegalArgumentException(
```

Comment:
What happens when you're aggregating over a data stream that contains both old and new indices? Will this return a partial failure? How does Kibana/Lens handle the failure?

Reply:
I don't know. I will look for a way to add an integration test for that

Reply:
Since it's a general question about aggregations on a field in a data stream, I guess we can add a test that is not related to the `_ignored` field. 
Something like:
1. create a component template that has a mapping for field `foo` with `"doc_values": false`
2. create a data stream index template that uses this mapping
3. index a document
5. update the component template so that the mapping for field `foo` is changed to `"doc_values": true`
6. rollover the data stream
7. index another document
8. check aggregation on field `foo`

Would that cover it?

Reply:
you would indeed get partial failures in this case, one failure for each shard belonging to the old indices, as they can't be aggregated on.